pysindy/feature_library/parameterized_library.py

```diff
@@ -0,0 +1,121 @@
+        a feature library. Value determined by num_parameters and num_features.
+
+    n_input_features_ : int
+        The total number of input features.
+        WARNING: This is deprecated in scikit-learn version 1.0 and higher so
+        we check the sklearn.__version__ and switch to n_features_in if needed.
+
+    n_output_features_ : int
+        The total number of output features. The number of output features
+        is the sum of the numbers of output features for each of the
+        concatenated libraries.
+
+    Examples
+    ----------
+    >>> import numpy as np
+    >>> from pysindy.feature_library import ParameterizedLibrary,PolynomialLibrary
+    >>> from pysindy import AxesArray
+    >>> xs=[np.random.random((5,3)) for n in range(3)]
+    >>> us=[np.random.random((5,3)) for n in range(3)]
+    >>> feature_lib=PolynomialLibrary(degree=3)
+    >>> parameter_lib=PolynomialLibrary(degree=1)
+    >>> lib=ParameterizedLibrary(feature_library=feature_lib,
+    >>>     parameter_library=parameter_lib,num_features=3,num_parameters=3)
+    >>> xus=[AxesArray(np.concatenate([xs[i],us[i]],axis=-1)) for i in range(3)]
+    >>> lib.fit(xus)
+    >>> lib.transform(xus)
+    """
+
+    def __init__(
+        self,
+        parameter_library=PolynomialLibrary(degree=1, include_bias=True),
+        feature_library=PolynomialLibrary(),
+        num_parameters=3,
+        num_features=3,
+        library_ensemble=False,
+        ensemble_indices=[0],
+    ):
+        if not isinstance(feature_library, BaseFeatureLibrary) or not isinstance(
+            parameter_library, BaseFeatureLibrary
+        ):
+            raise ValueError(
+                "Both feature_library and parameter_library must be instances of \
+                 BaseFeatureLibrary."
+            )
+
+        if num_parameters <= 0 or num_features <= 0:
+            raise ValueError("Both num_parameter and num_feature must be positive.")
+        libraries = [parameter_library, feature_library]
+        tensor_array = [[1, 1]]
+
+        feature_input = np.zeros(num_features + num_parameters, dtype=np.int32)
+        feature_input[:num_features] = np.arange(num_features)
+
+        parameter_input = (
+            np.ones(num_features + num_parameters, dtype=np.int32) * num_features
+        )
+        parameter_input[-num_parameters:] = num_features + np.arange(num_parameters)
+
+        inputs_per_libraries = np.array([parameter_input, feature_input])
+
+        super(ParameterizedLibrary, self).__init__(
+            libraries,
+            tensor_array=tensor_array,
+            exclude_libraries=[0, 1],
+            inputs_per_library=inputs_per_libraries,
+            library_ensemble=library_ensemble,
+            ensemble_indices=ensemble_indices,
+        )
+
+    def calc_trajectory(self, diff_method, x, t):
+        # if tensoring weak libraries, add the correction
+        if hasattr(self.libraries_[0], "K"):
```

Comment:
NIT: Mild preference for an `isinstance` check rather than `hasattr`, as it obviates the need for the comment.

Reply:
Yeah, I agree, but there are also some edge cases that need debugging here though if people try to do exotic things. If the first library here is a `ConcatLibrary` consisting of `WeakPDELibrary`'s, it should still add this correction...we could create a `self.K` in the weak case inside of `ConcatLibrary` and `TensoredLibrary`, so that this would be true then.  I think I will implement that in my next push, in addition to these other things.

Reply:
Ohhh, that makes sense, I think.  It probably is better as is, without adding a `K` attribute in to `ConcatLibrary`/`TensoredLibrary`, since that would break type stability.

Reply:
I'm adding a recursive `has_weak` and `has_nonweak`, and also a better `get_spatial_grid` in my next push, to resolve this issue and use `isinstance`.